### ISSUE TITLE
[Bug] Crash if entering PIP on Samsung devices with Talkback on

### ISSUE THREAD

**Issue by Mugurell** [mozilla-mobile/fenix#17966]
[comment]: # (Please do your best to search for duplicate issues before filing a new issue so we can keep our issue board clean)
[comment]: # (Every issue should have the exact bug and steps to reproduce described in it. Please do not file feedback list tickets as it is difficult to parse them and address their individual points)
[comment]: # (Read https://github.com/mozilla-mobile/fenix#i-want-to-file-an-issue for more information)

### Steps to reproduce
- On a Samsung device turn on Talkback
- In Fenix start playing a fullscreen video
- Press the home button to trigger entering in PIP

### Expected behavior
User gets to the homescreen with the video playing in a small floating window

### Actual behavior
Crash with the following stacktrace
 > java.lang.IllegalStateException: enterPictureInPictureMode: Device doesn't support picture-in-picture mode.
        at com.android.server.am.ActivityManagerService.ensureValidPictureInPictureActivityParamsLocked(ActivityManagerService.java:10646)
        at com.android.server.am.ActivityManagerService.enterPictureInPictureMode(ActivityManagerService.java:10519)
        at android.app.IActivityManager$Stub.onTransact(IActivityManager.java:3826)
        at com.android.server.am.ActivityManagerService.onTransact(ActivityManagerService.java:3863)

### Device information

* Device vendor / model and Android version: Samsung devices. Any Android version.
* Firefox for Android version: All Fenix versions are affected.
 
 ### Note
PIP functions as expected on Samsung devices if Talkback is off.

**Comment by Mugurell** [comment]
Proposed fix: Check beforehand and avoid trying to enter PIP on a Samsung device if an accessibility service is active.

**Comment by yichiehc** [comment]
Hi @eliserichards,
May I work on this issue?

**Comment by Leena1666** [comment]
can I work on this issue?

**Comment by Mugurell** [comment]
Assigning @yichiehc as they requested first.

**Comment by jonalmeida** [comment]
That's interesting. It looks like we were already trying to catch this same bug in the feature: https://github.com/mozilla-mobile/android-components/blob/master/components/feature/session/src/main/java/mozilla/components/feature/session/PictureInPictureFeature.kt#L51-L56

But it doesn't work?

**Comment by Mugurell** [comment]
Thank you @jonalmeida !
Seems like the exception is caught as expected in the code you've linked.
I just looked at the logcat and what threw me off was that ActivityManager would log with error level the exception shown in the ticket description but then the exception will bubble up and be caught in AC where it'd log it again with a warning level but also prevent it crashing the app.
Seems like there is nothing to do here. Sorry for the confusion. Will close this.

**Comment by jonalmeida** [comment]
@Mugurell we could fix the logging in the feature in that case? It might make more sense to clarify that the feature caught it, but we don't need to throw the exception into the logs again.

We are already sending to our crash reporting either way to monitor.

**Comment by Mugurell** [comment]
For me the issue is that ActivityManager will log the crash with error level before we caught it. And seems like this cannot be avoided.
(Other than checking for samsung && a11y before trying to enter pip but I saw that comment about this not being enough)
In that problematic scenario pressing home would not put the app in PiP and there would also be an exception logged in logcat so maybe others just like me would think that the app really crashed.

I don't know if that crash would be logged by ActivityMonitor on all devices though and since we log that it a lower priority afterwards maybe it's ok to stay just as a backup log? If that ain't broken..
